### PR TITLE
Fix ReadAndSanitizeHeaderEntry method

### PR DESCRIPTION
**Pull request by ddscher**
Add position information when examining content buffer to prevent reading at irregular buffer position
Add unit test to test `ReadAndSanitizeHeaderEntry `method
Add additional test data

### CODE REVIEW

File: src/Formplot/FileFormat/FormplotHelper.cs [hunk 1]
```diff
@@ -93,9 +93,12 @@ private static bool IsTruncationSafe( byte[] content )
...
 
-		private static int FindEndOfContent( byte[] content )
+		private static int FindEndOfContent( byte[] content, int position )
 		{
-			var i = content.Length - 1;
+			if( position < 0 || position > content.Length - 1 )
+			 	return content.Length - 1;
+
+			var i = position;
 			while( i > -1 && content[ i ] == 0 )
```
Comment (georg-eckert-zeiss) [inline src/Formplot/FileFormat/FormplotHelper.cs:102]: What do the zeros mean? Is the end of the content array filled up with 0?

File: src/Formplot/FileFormat/FormplotHelper.cs [hunk 2]
```diff
@@ -93,9 +93,12 @@ private static bool IsTruncationSafe( byte[] content )
 					&& ( content[ 4 ] == 0x6C );
 		}
 
-		private static int FindEndOfContent( byte[] content )
+		private static int FindEndOfContent( byte[] content, int position )
 		{
-			var i = content.Length - 1;
+			if( position < 0 || position > content.Length - 1 )
```
[the remark this answers is not in the document]
Comment (ddscher) [reply src/Formplot/FileFormat/FormplotHelper.cs:98]: We could throw an `ArgumentOutOfRangeException` instead. What do you think?

File: src/Formplot/FileFormat/FormplotHelper.cs (hunk 1 above)
Comment (ddscher) [reply src/Formplot/FileFormat/FormplotHelper.cs:102]: This code wasn't touched by me. I think, it's even your code.

Comment (georg-eckert-zeiss) [reply src/Formplot/FileFormat/FormplotHelper.cs:102]: It's not. But it's fine then.

**Review by georg-eckert-zeiss — APPROVED** [review]
Should be fine.

**Pull request merged**